vhdbuilder/packer/backfill-cleanup.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -x
+
+EXPIRATION_IN_HOURS=168
+# convert to seconds so we can compare it against the "tags.now" property in the resource group metadata
+(( expirationInSecs = ${EXPIRATION_IN_HOURS} * 60 * 60 ))
+# deadline = the "date +%s" representation of the oldest age we're willing to keep
+(( deadline=$(date +%s)-${expirationInSecs%.*} ))
+
+# we will clean old sig image-definitions/versions/galleries, RGs only if 
+# we have overriden the pipeline variable SKIP_IMAGE_DELETION_BACKFILL to False
+# or we are running on production pipelines
+CLEAN_OLD_WINDOWS_RESOURCES="False"
+if [[ "${SKIP_IMAGE_DELETION_BACKFILL}" == "False" || "${DRY_RUN}" == "False" ]]; then
+  CLEAN_OLD_WINDOWS_RESOURCES="True"
```

Comment:
could we exit with 0 if we do not need to cleanup the old resources?

Reply:
Since this is only used by Windows, it may be better to exit with 0 when MODE is not windowsVhdMode

Reply:
yeah think this would be cleaner if we just say:

`
[[ "${SKIP_IMAGE_DELETION_BACKFILL"} == "True" ]] && [[ "${DRY_RUN}" == "True" ]] && exit 0
`

then we don't need `CLEAN_OLD_WINDOWS_RESOURCES` at all, unless I'm missing something.